torch_xla/csrc/tensor_methods.cpp

```diff
@@ -2639,6 +2641,21 @@ XLATensorPtr view(const XLATensorPtr& input,
   return input->CreateViewTensor(std::move(view_info));
 }
 
+// Here we know which output shape dimension is dynamic.
+XLATensorPtr view_symint(const XLATensorPtr& input,
+                         at::SymIntArrayRef sym_size) {
+  xla::util::MaybeRef<xla::Shape> input_shape = input->shape();
+  SymIntElements size_elements(sym_size);
+  std::vector<int64_t> complete_dimensions = GetCompleteShape(
+      size_elements.GetUpperBounds(), input_shape.get().dimensions());
+  xla::Shape result_shape = xla::ShapeUtil::MakeShape(
+      input_shape.get().element_type(), complete_dimensions,
+      size_elements.GetDynamicDims());
+  ViewInfo view_info(ViewInfo::Type::kReshape, std::move(result_shape),
+                     input_shape);
+  return input->CreateViewTensor(std::move(view_info));
```

Comment:
hmm.. so reshape handles dynamism out of the box? I guess that make sense.

Reply:
Even so, I'd wonder if we need to verify this with a unit test to confirm. Ideally, all ops we touch should have a 'simple' unit test to be whitelisted for dynamism - we can even have a whitelisting `yaml` file that tracks support level. For ops we do proper dynamism sypport, we do more extensive testing. With this approach, future regressions will less likely become a surprise to us. Thoughts?

Reply:
+1 on adding unit tests. For this PR, I have verified this path with 4 unit tests in pytorch/xla/test/test_dynamic_shapes.py in this PR. Let me know if I have missed any scenarios. Also, for all ops that support dynamism, I have added enough unit tests (the remaining untested code are the Size nodes which I will add tests this week.)